### PR TITLE
Recurring payment badge on the gateways list

### LINKED ISSUE #53384: [WC Payments NOX] Support recurring payment badge on the gateways list
### Description
We need to support recurring payments badge on the gateways list

Image

### Designs
Design post: pfHfG4-q7-p2
Figma: xO8YdpKfaVoP2d8NvweZ5I-fi-1428_21437

### Dev notes
Reactifying the Payments settings page: pe4Hk0-1ly-p2
Populate react-powered main payments settings screen PR: https://github.com/woocommerce/woocommerce/pull/50825

### Additional context
PRD: 1wMfX8CAjDnOp24tcU-PADB0pOt3cz7mGdyZYeRBxx2Y-gdoc

### PR DESCRIPTION
**Pull request by elazzabi**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This adds a "Recurring payments" icon next to providers offering that.

Closes #53384

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Pre-requisites for local testing:**

- Have woocommerce installed locally and synced within WooPayments. See Pe4Hk0-3f-p2 for details.
- Enable the feature flag to see reactified settings page. pe4Hk0-1ly-p2.
- Check out this branch.
- Run pnpm i && pnpm run build

**Testing instructions:**

1. First, you need **WooCommerce Subscriptions** to test this. Please get the zip and install it locally.
1. Navigate to WooCommerce > Settings > Payments
2. Select a country where WooPayments is available (UAE for me)
3. You should see a new "Recurring payments" icon next to WooPayments

Screenshot 2024-12-12 at 16 25 34

4. On hovering, a tooltip should be displayed:

Screenshot 2024-12-12 at 16 33 02

5. Go to "Other payments options" and install Stripe
6. Once installed, the same icon and tooltip should be displayed too

Screenshot 2024-12-12 at 16 35 09

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add "Recurring payments" icon next to payment providers

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/changelog/53669-add-add-recurring-payments-icon [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: add
+
+Add "Recurring payments" icon next to payment providers
```
Comment (anu-rock) [inline plugins/woocommerce/changelog/53669-add-add-recurring-payments-icon:4]:
```suggestion
Add a "Recurring payments" icon next to installed payment providers that support WooCommerce Susbscriptions.
```

Comment (anu-rock) [reply plugins/woocommerce/changelog/53669-add-add-recurring-payments-icon:4]: Newline character is missing at the end.

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx [hunk 2]
```diff
@@ -69,6 +71,26 @@ export const PaymentGatewayListItem = ( {
...
+									'woocommerce'
+								) }
+								children={
+									<img
+										src={
+											WC_ASSET_URL +
+											'images/icons/recurring-payments.svg'
+										}
+										alt={ __(
+											'Recurring payments',
```
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx]:
```suggestion
											'Icon to indicate support for recurring payments',
```

**Review by anu-rock — APPROVED** [review]
___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ⚠️  Missing, nice to have |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | Not applicable |
| Changelog  | 👍  Almost good (left a comment) |
| Documentation  | Not applicable |
| PR description  | ✅ All good |
| Testing instructions  | ✅ All good |
| Manual testing  | ✅ Works as expected |

#### Scenarios tested

| Scenario  | Result | Screenshot |
| ------------- | ------------- | ------------- |
| WooPayments, Stripe, PayPal installed; WooCommerce Subscriptions installed and active  | ✅ Recurring icon displayed for all three payment providers | Screenshot 2024-12-13 at 1 12 57 PM |
| WooPayments, Stripe, PayPal installed; WooCommerce Subscriptions installed but deactivated | ✅ Recurring icon not displayed for any of the three payment providers | Screenshot 2024-12-13 at 1 14 25 PM |

File: plugins/woocommerce/changelog/53669-add-add-recurring-payments-icon (hunk 1 above)
Comment (oaratovskyi) [reply plugins/woocommerce/changelog/53669-add-add-recurring-payments-icon:4]: It's added automatically, I've tried to remove it in the other pr - then CI removes it back 🤷

**Review by oaratovskyi — APPROVED** [review]
Changed LGTM and it worked well 👍

**Pull request merged**